src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -389,6 +449,28 @@ export class DeploymentsService implements OnDestroy {
       });
   }
 
+  private getTimeConstrainedTimeseriesData(spaceId: string, applicationId: string, environmentName: string, startTime: number, endTime: number): Observable<MultiTimeseriesData> {
+    const emptyResult: MultiTimeseriesData = {
+      cores: [],
+      memory: [],
+      net_tx: [],
+      net_rx: [],
+      start: startTime,
+      end: endTime
+    };
+    return this.isApplicationDeployedInEnvironment(spaceId, applicationId, environmentName)
+    .flatMap((deployed: boolean) => {
+      if (deployed) {
+        const url = `${this.apiUrl}${spaceId}/applications/${applicationId}/deployments/${environmentName}/statseries?start=${startTime}&end=${endTime}`;
+        return this.http.get(url, { headers: this.headers })
+          .map((response: Response) => (response.json() as MultiTimeseriesResponse).data)
+          .catch(() => Observable.of(emptyResult));
```

Comment:
Is there a particular reason for this emptyResult behaviour for this request, or is it just because it's been rebased since before handleHttpError was introduced? The other HTTP requests in this service now handle the case of a failed request by emitting an error event and logging the error, rather than returning an empty default value.

Reply:
This is a relatively larger change so following up in a separate commit to bring the behaviour in line is fine.

Reply:
Yeah I was working on a branch that was before handleHttpError was introduced, so I updated it in a second commit.